zrml/prediction-markets/src/lib.rs

```diff
@@ -208,22 +262,27 @@ mod pallet {
         ///
         /// NOTE: This is the only way to create new shares.
         ///
-        #[pallet::weight(50_000_000)]
+        // Note: buy_complete_sets weight consumption is dependant on how many assets exists
+        // Unfortunately this information can only be retrieved with a storage call, therefore
+        // The worst-case scenario is assumed and the correct weight is calculated at the end of this function.
+        // This also occurs in numereous other functions
```

Comment:
```suggestion
        // This also occurs in numerous other functions.
```

style => fixed typos
content => makes sense